crypto/evp/keymgmt_lib.c

```diff
@@ -118,19 +92,62 @@ void *evp_keymgmt_export_to_provider(EVP_PKEY *pk, EVP_KEYMGMT *keymgmt,
                  * forgets to check the return value.
 
                  */
-                if (exportfn(pk->pkeys[j].provdata, &try_import, &import_data)
+                if (exportfn(pk->pkeys[i].provdata, &try_import, &import_data)
                     && (provdata = import_data.provdata) != NULL)
                     break;
             }
         }
+
+        /*
+         * TODO(3.0) Right now, we assume we have ample space.  We will
+         * have to think about a cache aging scheme, though, if |i| indexes
+         * outside the array.
+         */
+        i = ossl_assert(i < OSSL_NELEM(pk->pkeys));
```

Comment:
What is the purpose of the assignment to `i` here?

Reply:
Must be old cruft.  Removing.

Reply:
Nope, but if you read the comment a couple of lines up, I think the reason will become clear :wink:

Reply:
Wouldnt if (!ossl_assert())
                   return 0;
be better?